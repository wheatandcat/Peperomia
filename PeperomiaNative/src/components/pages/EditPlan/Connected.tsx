import * as SQLite from 'expo-sqlite';
import React, {
  useContext,
  useState,
  memo,
  useEffect,
  useCallback,
} from 'react';
import { NavigationScreenProp, NavigationRoute } from 'react-navigation';
import { Alert } from 'react-native';
import {
  Context as ItemsContext,
  ContextProps as ItemContextProps,
} from '../../../containers/Items';
import { db, ResultError } from '../../../lib/db';
import { update as updateItem, Item } from '../../../lib/db/item';
import {
  update as updateCalendar,
  insert as insertCalendar,
  Calendar,
} from '../../../lib/db/calendar';
import getKind from '../../../lib/getKind';
import { SuggestItem } from '../../../lib/suggest';
import { useDidMount } from '../../../hooks/index';
import Page from '../../templates/CreatePlan/Page';

type Props = {
  navigation: NavigationScreenProp<NavigationRoute>;
};

type ConnectedProps = Pick<
  ItemContextProps,
  'items' | 'refreshData' | 'calendars'
> & {
  navigation: NavigationScreenProp<NavigationRoute>;
};

type PlanProps = Pick<ItemContextProps, 'items' | 'refreshData'> & {
  navigation: NavigationScreenProp<NavigationRoute>;
  input: {
    title: string;
    date: string;
  };
  image: string;
  kind: string;
  onInput: (name: string, value: any) => void;
  onImage: (image: string) => void;
  onSave: () => void;
  onIcons: () => void;
  onHome: () => void;
};

type State = {
  input: {
    title: string;
    date: string;
  };
  image: string;
  kind: string;
  calendar: Calendar;
};

export default (props: Props) => {
  const { refreshData, items, calendars } = useContext(ItemsContext);

  return (
    <Connected
      navigation={props.navigation}
      items={items}
      refreshData={refreshData}
      calendars={calendars}
    />
  );
};

const Connected = memo((props: ConnectedProps) => {
  const [state, setState] = useState<State>({
    input: { title: '', date: '' },
    image: '',
    kind: '',
    calendar: {
      id: 0,
      itemId: 0,
      date: '',
    },
  });

  useDidMount(() => {
    const id = props.navigation.getParam('id', 0);
    let input: {
      date: string;
      title: string;
    } = {
      date: '',
      title: '',
    };

    const calendar = (props.calendars || []).find(
      item => Number(id) === Number(item.itemId)
    );

    if (calendar && calendar.date) {
      input.date = calendar.date;
      setState(s => ({
        ...s,
        ...state,
        calendar,
      }));
    }

    const schedule = (props.items || []).find(
      item => Number(id) === Number(item.id)
    );

    if (schedule && schedule.title) {
      input.title = schedule.title;
    }

    const image = props.navigation.getParam('image', '');
    const kind = props.navigation.getParam('kind', '');

    setState(s => ({
      ...s,
      input,
      image,
      kind,
    }));
  });

  useEffect(() => {
    const image = props.navigation.getParam('image', '');
    if (image && image !== state.image) {
      setState(s => ({
        ...s,
        image,
      }));
    }

    const kind = props.navigation.getParam('kind', '');
    if (kind && kind !== state.kind) {
      setState(s => ({
        ...s,
        kind,
      }));
    }
  }, [props.navigation, state]);

  const onImage = useCallback((image: string) => {
    setState(s => ({
      ...s,
      image,
    }));
  }, []);

  const onInput = useCallback(
    (name: string, value: any) => {
      const input = {
        ...state.input,
        [name]: value,
      };

      setState(s => ({
        ...s,
        input,
      }));
    },
    [state]
  );

  const save = useCallback(
    (_: Item[], error: ResultError) => {
      if (error) {
        Alert.alert('保存に失敗しました');
        return;
      }

      const id = props.navigation.getParam('id', 0);

      props.navigation.navigate('Schedule', {
        itemId: id,
        title: state.input.title,
      });
    },
    [props.navigation, state.input.title]
  );

  const saveCalendar = useCallback(
    (_: Calendar | number, error: ResultError) => {
      if (error) {
        return;
      }

      if (props.refreshData) {
        props.refreshData();
      }
    },
    [props]
  );

  const onSave = useCallback(async () => {
    if (state.input.date) {
      const check = (props.calendars || []).find(
        calendar => calendar.date === state.input.date
      );

      if (check) {
        Alert.alert('同じ日にスケジュールが既に登録されています');
        return;
      }
    }

    db.transaction((tx: SQLite.Transaction) => {
      const id = props.navigation.getParam('id', 0);

      const item: Item = {
        id,
        title: state.input.title,
        kind: state.kind || getKind(state.input.title),
        image: '',
      };

      updateItem(tx, item, save);

      if (!state.input.date) {
        return;
      }

      if (state.calendar.id) {
        updateCalendar(
          tx,
          {
            ...state.calendar,
            date: state.input.date,
          },
          saveCalendar
        );
      } else {
        insertCalendar(
          tx,
          {
            itemId: id,
            date: state.input.date,
          },
          saveCalendar
        );
      }
    });
  }, [
    props.calendars,
    props.navigation,
    save,
    saveCalendar,
    state.calendar,
    state.input.date,
    state.input.title,
    state.kind,
  ]);

  const onIcons = useCallback(() => {
    props.navigation.navigate('Icons', {
      kind: getKind(state.input.title),
      onSelectIcon: (kind: string) => {
        props.navigation.navigate('EditPlan', {
          kind: kind,
        });
      },
      onDismiss: () => {
        props.navigation.navigate('EditPlan');
      },
      photo: true,
    });
  }, [props.navigation, state.input.title]);

  const onHome = useCallback(() => {
    props.navigation.goBack();
  }, [props.navigation]);

  console.log(state);

  return (
    <Plan
      navigation={props.navigation}
      input={state.input}
      image={state.image}
      kind={state.kind}
      items={props.items}
      refreshData={props.refreshData}
      onInput={onInput}
      onImage={onImage}
      onSave={onSave}
      onIcons={onIcons}
      onHome={onHome}
    />
  );
});

type PlanState = {
  suggestList: SuggestItem[];
};

const Plan = memo((props: PlanProps) => {
  const [state, setState] = useState<PlanState>({
    suggestList: [],
  });

  useDidMount(() => {
    const suggestList = (props.items || []).map(item => ({
      title: item.title,
      kind: item.kind,
    }));

    setState(s => ({
      ...s,
      suggestList,
    }));
  });

  const onSave = useCallback(async () => {
    await props.onSave();

    if (props.refreshData) {
      props.refreshData();
    }
  }, [props]);

  return (
    <Page
      mode="edit"
      title={props.input.title}
      date={props.input.date}
      image={props.image}
      kind={props.kind}
      suggestList={state.suggestList}
      onInput={props.onInput}
      onImage={props.onImage}
      onSave={onSave}
      onIcons={props.onIcons}
      onHome={props.onHome}
    />
  );
});
