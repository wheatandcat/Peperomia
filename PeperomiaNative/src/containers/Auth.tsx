import * as GoogleSignIn from "expo-google-sign-in";
import * as Google from "expo-google-app-auth";
import { AsyncStorage, Platform } from "react-native";
import React, { createContext, Component } from "react";
import Constants from "expo-constants";
import { Sentry } from "react-native-sentry";
import * as firebase from "firebase";

const Context = createContext({});
const { Provider } = Context;

interface Props {}

interface State {
  uid: string;
  email: string;
}

const isStandaloneAndAndroid = () => {
  return Platform.OS === "android" && Constants.appOwnership !== "expo";
};

export default class extends Component<Props, State> {
  state = {
    email: "",
    uid: ""
  };

  async componentDidMount() {
    if (isStandaloneAndAndroid()) {
      const androidClientId = process.env.GOOGLE_LOGIN_ANDROID_CLIENT_ID;
      try {
        await GoogleSignIn.initAsync({
          clientId: String(androidClientId)
        });
      } catch ({ message }) {
        Sentry.captureMessage(JSON.stringify(message));
      }
    }

    const loggedIn = await this.loggedIn();

    if (loggedIn && !this.state.uid) {
      const uid = await AsyncStorage.getItem("uid");
      if (uid) {
        this.setState({
          uid
        });
      }
    }

    if (loggedIn && !this.state.email) {
      const email = await AsyncStorage.getItem("email");
      if (email) {
        this.setState({
          email
        });
      }
    }
  }

  onGoogleLogin = async () => {
    if (isStandaloneAndAndroid()) {
      // TODO: AndroidのstandaloneのみGoogleSignInを使わないとエラーになる
      // https://github.com/expo/expo/issues/4762

      await GoogleSignIn.askForPlayServicesAsync();
      const result = await GoogleSignIn.signInAsync();

      if (result.type === "success" && result.user && result.user.auth) {
        const { idToken, accessToken } = result.user.auth;
        await this.firebaseLogin(idToken || "", accessToken || "");
      } else {
        Sentry.captureMessage(JSON.stringify(result));
      }
    } else {
      const androidClientId = process.env.GOOGLE_LOGIN_ANDROID_CLIENT_ID;
      const iosClientId = process.env.GOOGLE_LOGIN_IOS_CLIENT_ID;
      const result = await Google.logInAsync({
        clientId:
          Platform.OS === "ios" ? String(iosClientId) : String(androidClientId),
        iosClientId,
        androidClientId,
        scopes: ["profile", "email"]
      });

      if (result.type === "success") {
        const { idToken, accessToken } = result;
        await this.firebaseLogin(idToken || "", accessToken || "");
      } else {
        Sentry.captureMessage(JSON.stringify(result));
      }
    }
  };

  firebaseLogin = async (idToken: string, accessToken: string) => {
    const credential = firebase.auth.GoogleAuthProvider.credential(
      idToken,
      accessToken
    );

    await firebase
      .auth()
      .signInAndRetrieveDataWithCredential(credential)
      .catch(error => {
        Sentry.captureMessage(JSON.stringify(error));
      });

    await this.setSession(true);
  };

  loggedIn = async () => {
    const idToken = await this.getIdToken();

    return Boolean(idToken);
  };

  logout = async () => {
    await firebase.auth().signOut();

    await AsyncStorage.removeItem("id_token");
    await AsyncStorage.removeItem("expiration");
    await AsyncStorage.removeItem("email");
  };

  setSession = async (refresh = false) => {
    const user = firebase.auth().currentUser;
    if (!user) {
      return;
    }

    if (user.email) {
      await AsyncStorage.setItem("email", user.email);
      await AsyncStorage.setItem("uid", user.uid);
      this.setState({
        email: user.email,
        uid: user.uid
      });
    }

    const idToken = await user.getIdToken(refresh);
    await AsyncStorage.setItem("id_token", idToken);
    await AsyncStorage.setItem(
      "expiration",
      String(new Date().getTime() + 60 * 60)
    );

    return idToken;
  };

  getIdToken = async () => {
    const idToken = await AsyncStorage.getItem("id_token");
    if (!idToken) {
      return null;
    }

    const expiration = await AsyncStorage.getItem("expiration");
    if (Number(expiration) > new Date().getTime()) {
      return idToken;
    }

    return this.setSession(true);
  };

  render() {
    return (
      <Provider
        value={{
          onGoogleLogin: this.onGoogleLogin,
          getIdToken: this.getIdToken,
          loggedIn: this.loggedIn,
          logout: this.logout,
          email: this.state.email,
          uid: this.state.uid
        }}
      >
        {this.props.children}
      </Provider>
    );
  }
}

export const Consumer = Context.Consumer;
