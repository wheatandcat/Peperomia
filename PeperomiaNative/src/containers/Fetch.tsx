import React, { createContext, Component, ReactNode } from 'react';
import Constants from 'expo-constants';
import {
  Consumer as AuthConsumer,
  ContextProps as AuthContextProps,
} from './Auth';

export const Context = createContext<ContextProps>({});
const { Provider } = Context;

export type ContextProps = Partial<Pick<Connected, 'post'>>;

type Props = {
  children: ReactNode;
};

export default class extends Component<Props> {
  render() {
    return (
      <AuthConsumer>
        {({ getIdToken }: AuthContextProps) => (
          <Connected getIdToken={getIdToken}>{this.props.children}</Connected>
        )}
      </AuthConsumer>
    );
  }
}

type ConnectedProps = Pick<AuthContextProps, 'getIdToken'>;

export interface FetchValue {
  post: (url: string, param: any) => Promise<Response>;
}

class Connected extends Component<ConnectedProps> {
  post = async (url: string, param: any): Promise<Response> => {
    const idToken = this.props.getIdToken ? await this.props.getIdToken() : '';
    const apiHost = process.env.API_HOST;
    const init = {
      ...param,
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${idToken}`,
      },
    };

    if (!Constants.isDevice) {
      console.info(init);
    }

    console.info(`Bearer ${idToken}`);

    const response = await fetch(`${apiHost}/${url}`, init);
    console.log(response);
    if (!Constants.isDevice) {
      if (!response.ok) {
        const result = await response.json();
        console.info(result);
      }
    }

    return response;
  };

  render() {
    return (
      <Provider
        value={{
          post: this.post,
        }}
      >
        {this.props.children}
      </Provider>
    );
  }
}

export const Consumer = Context.Consumer;
