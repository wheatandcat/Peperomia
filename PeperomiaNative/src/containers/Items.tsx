import { SQLite } from "expo-sqlite";
import React, { createContext, Component } from "react";
import { db } from "../lib/db";
import { select as selectItems, Item } from "../lib/db/item";
import {
  selectByItemId as selectItemDetailByItemId,
  ItemDetail
} from "../lib/db/itemDetail";

const Context = createContext({});
const { Provider } = Context;

interface ItemAbout {
  itemId: number;
  about: string;
}

interface Props {}

interface State {
  loading: boolean;
  items: Item[];
  about: ItemAbout[];
}

export default class extends Component<Props, State> {
  state = {
    loading: true,
    items: [],
    about: []
  };

  componentDidMount() {
    this.getData();
  }

  getData = () => {
    db.transaction((tx: SQLite.Transaction) => {
      selectItems(tx, this.setItems);
    });
  };

  setItems = (data: any, error: any) => {
    if (error) {
      return;
    }
    this.setState({
      items: data,
      about: []
    });

    data.map((val: Item) => {
      db.transaction((tx: SQLite.Transaction) => {
        selectItemDetailByItemId(tx, String(val.id), this.setItemsDetail);
      });
    });
  };

  setItemsDetail = (data: any, error: any) => {
    if (error || !data || data.length === 0) {
      return;
    }

    const names = data.map((val: ItemDetail) => val.title).join("â†’");
    const itemId = data[0].itemId;
    const about = [
      ...this.state.about,
      {
        itemId: itemId,
        about: names
      }
    ];

    this.setState({
      about,
      loading: false
    });
  };

  render() {
    return (
      <Provider
        value={{
          items: this.state.items,
          about: this.state.about,
          refreshData: this.getData,
          itemsLoading: this.state.loading
        }}
      >
        {this.props.children}
      </Provider>
    );
  }
}

export const Consumer = Context.Consumer;
