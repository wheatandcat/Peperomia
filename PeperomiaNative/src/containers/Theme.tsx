import React, { createContext, Component } from "react";
import { AsyncStorage } from "react-native";
import Spinner from "react-native-loading-spinner-overlay";
import { setMode } from "../config/theme";

const Context = createContext({});
const { Provider } = Context;

type Props = {};

type State = {
  rerendering: boolean;
  mode: string;
  render: boolean;
};

export type ContextProps = Partial<
  Pick<State, "rerendering" | "mode"> & {
    onModeChange: (mode: "light" | "dark") => void;
    onFinishRerendering: () => void;
  }
>;

export default class extends Component<Props, State> {
  state = {
    rerendering: false,
    mode: "light",
    render: false
  };

  async componentDidMount() {
    let mode = await AsyncStorage.getItem("THEME_MODE");
    console.log(mode);

    if (mode !== "dark") {
      mode = "light";
    }

    if (mode === "light" || mode === "dark") {
      setMode(mode);
    }

    this.setState({
      mode,
      render: true
    });
  }

  onModeChange = async (mode: "light" | "dark") => {
    this.setState({
      mode,
      rerendering: true
    });

    await AsyncStorage.setItem("THEME_MODE", mode);

    if (mode === "light" || mode === "dark") {
      setMode(mode);
      setTimeout(this.setUnRender.bind(this), 100);
      setTimeout(this.setRender.bind(this), 101);
    }
  };

  setUnRender = () => {
    this.setState({
      render: false
    });
  };

  setRender = () => {
    this.setState({
      render: true
    });
  };

  onFinishRerendering = () => {
    this.setState({
      rerendering: false
    });
  };

  render() {
    return (
      <>
        <Spinner visible={this.state.rerendering} textContent="" />
        {this.state.render && (
          <Provider
            value={{
              rerendering: this.state.rerendering,
              mode: this.state.mode,
              onModeChange: this.onModeChange,
              onFinishRerendering: this.onFinishRerendering
            }}
          >
            {this.props.children}
          </Provider>
        )}
      </>
    );
  }
}

export const Consumer = Context.Consumer;
