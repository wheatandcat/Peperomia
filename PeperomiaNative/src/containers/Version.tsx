import { Component } from "react";
import { AsyncStorage } from "react-native";
import compareVersions from "compare-versions";
import { migrationV104 } from "../lib/migration";

interface Props {}

interface State {
  loading: boolean;
}

export default class extends Component<Props, State> {
  state = {
    loading: true
  };

  async componentDidMount() {
    let appVerion = await AsyncStorage.getItem("APP_VERSION");
    if (!appVerion) {
      appVerion = "1.0.0";
    }

    if (compareVersions.compare("1.0.4", appVerion, ">")) {
      // カラム追加のマイグレーション実行
      const ok = await migrationV104();
      if (ok) {
        // 現在のバージョンを設定
        AsyncStorage.setItem("APP_VERSION", "1.0.4");
      }
    }

    this.setState({
      loading: false
    });
  }

  render() {
    if (this.state.loading) {
      return null;
    }

    return this.props.children;
  }
}
