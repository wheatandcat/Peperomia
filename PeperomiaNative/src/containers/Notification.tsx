import React, { memo, createContext, FC, useCallback, useContext } from 'react';
import { Platform, Alert } from 'react-native';
import * as Notifications from 'expo-notifications';
import * as Permissions from 'expo-permissions';
import Constants from 'expo-constants';
import { useFetch } from 'containers/Fetch';

export const Context = createContext<ContextProps>({});
const { Provider } = Context;

export type ContextProps = Partial<{
  onPermissionRequest: () => Promise<boolean>;
}>;

type Props = {};

const Notification: FC<Props> = memo((props) => {
  const { post } = useFetch();

  const onPermissionRequest = useCallback(async () => {
    if (!post) {
      return false;
    }

    if (!Constants.isDevice) {
      Alert.alert('端末から実行してくだださい');
      return false;
    }

    const { status: existingStatus } = await Permissions.getAsync(
      Permissions.NOTIFICATIONS
    );

    let finalStatus = existingStatus;
    if (existingStatus !== Permissions.PermissionStatus.GRANTED) {
      const { status } = await Permissions.askAsync(Permissions.NOTIFICATIONS);
      finalStatus = status;
    }

    if (finalStatus !== 'granted') {
      Alert.alert('Push通知のトークンの取得に失敗しました');
      return false;
    }

    const { data: token } = await Notifications.getExpoPushTokenAsync();

    if (Platform.OS === 'android') {
      Notifications.setNotificationChannelAsync('default', {
        name: 'default',
        importance: Notifications.AndroidImportance.MAX,
        vibrationPattern: [0, 250, 250, 250],
        lightColor: '#FF231F7C',
      });
    }

    const request = {
      pushToken: {
        token,
        deviceId: Constants.deviceId,
      },
    };

    //Alert.alert(token);
    //console.log(token);

    const response = await post('CreatePushToken', request);

    if (response.error) {
      Alert.alert('更新に失敗しました');
      return false;
    }

    return true;
  }, [post]);

  return <Provider value={{ onPermissionRequest }}>{props.children}</Provider>;
});

export default Notification;

export const useNotification = () => useContext(Context);
export const Consumer = Context.Consumer;
