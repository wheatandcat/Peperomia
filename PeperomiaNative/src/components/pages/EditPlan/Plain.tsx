import React, { useState, memo, useCallback } from 'react';
import { SuggestItem } from 'lib/suggest';
import { useDidMount } from 'hooks/index';
import { ContextProps as ItemContextProps } from 'containers/Items';
import Page from 'components/templates/CreatePlan/Page';
import { Props } from './';

type PlainProps = Props &
  Pick<ItemContextProps, 'items' | 'refreshData'> & {
    input: {
      title: string;
      date: string;
    };
    kind: string;
    onInput: (name: string, value: any) => void;
    onSave: () => void;
    onIcons: () => void;
    onHome: () => void;
  };

type PlainState = {
  suggestList: SuggestItem[];
};

const Plain = memo((props: PlainProps) => {
  const [state, setState] = useState<PlainState>({
    suggestList: [],
  });

  useDidMount(() => {
    const suggestList = (props.items || []).map((item) => ({
      title: item.title,
      kind: item.kind,
    }));

    setState((s) => ({
      ...s,
      suggestList,
    }));
  });

  const onSave = useCallback(async () => {
    await props.onSave();

    if (props.refreshData) {
      props.refreshData();
    }
  }, [props]);

  return (
    <Page
      mode="edit"
      title={props.input.title}
      date={props.input.date}
      kind={props.kind}
      suggestList={state.suggestList}
      onInput={props.onInput}
      onSave={onSave}
      onIcons={props.onIcons}
      onHome={props.onHome}
    />
  );
});

export default Plain;
