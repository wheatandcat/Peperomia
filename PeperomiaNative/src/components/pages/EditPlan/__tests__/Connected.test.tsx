import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import * as Items from 'containers/Items';
import * as Auth from 'containers/Auth';
import EditPlan, { Connected } from '../Connected';

describe('components/pages/EditPlan/Connected.tsx', () => {
  describe('EditPlan', () => {
    let wrapper: ShallowWrapper;

    const propsData: any = () => ({
      title: 'test',
      kind: 'park',
      navigation: {
        setParams: jest.fn(),
        navigate: jest.fn(),
      },
      route: {
        params: {
          itemId: '1',
          refresh: '',
        },
      },
    });

    beforeEach(() => {
      jest.spyOn(Items, 'useItems').mockImplementation(
        () =>
          ({
            refreshData: jest.fn(),
            items: [],
            calendars: [],
          } as any)
      );

      wrapper = shallow(<EditPlan {...propsData()} />);
    });

    it('正常に表示されている', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Connected', () => {
    let wrapper: ShallowWrapper;

    const propsData: any = () => ({
      title: 'test',
      kind: 'park',
      navigation: {
        setParams: jest.fn(),
        navigate: jest.fn(),
      },
      route: {
        params: {
          itemId: '1',
          refresh: '',
        },
      },
      items: [],
      refreshData: jest.fn(),
      calendars: [],
    });

    beforeEach(() => {
      jest.spyOn(Auth, 'useAuth').mockImplementation(
        () =>
          ({
            uid: 'test',
          } as any)
      );
      jest.spyOn(React, 'useEffect').mockImplementation((f) => f());
      wrapper = shallow(<Connected {...propsData()} />);
    });

    it('正常に表示されている', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
