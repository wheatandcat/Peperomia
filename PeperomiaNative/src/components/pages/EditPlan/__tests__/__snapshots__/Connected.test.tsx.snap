// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/EditPlan/Connected.tsx Connected 正常に表示されている 1`] = `
<Memo()
  calendars={Array []}
  input={
    Object {
      "date": "",
      "title": "",
    }
  }
  items={Array []}
  kind=""
  navigation={
    Object {
      "navigate": [MockFunction],
      "setParams": [MockFunction],
    }
  }
  onHome={[Function]}
  onIcons={[Function]}
  onInput={[Function]}
  onSave={[Function]}
  refreshData={[MockFunction]}
  route={
    Object {
      "params": Object {
        "itemId": "1",
        "refresh": "",
      },
    }
  }
  title="test"
/>
`;

exports[`components/pages/EditPlan/Connected.tsx EditPlan 正常に表示されている 1`] = `
<Memo()
  calendars={Array []}
  items={Array []}
  kind="park"
  navigation={
    Object {
      "navigate": [MockFunction],
      "setParams": [MockFunction],
    }
  }
  refreshData={[MockFunction]}
  route={
    Object {
      "params": Object {
        "itemId": "1",
        "refresh": "",
      },
    }
  }
  title="test"
/>
`;
